### PR TITLE
Add Portuguese language support to client

### PR DESCRIPTION
**Pull request by davidklaw**
This is a continuation of #656, which appears to be a standstill.

I fixed the compilation errors and added or revised some of the translations if missing or ill-fitting. I adjusted layouts where shorter translations weren't possible or felt undesirable.

For visual comparison, I compiled these sheets of English vs. Portuguese:

image
image
image
image
image
image

This closes #656

### CODE REVIEW

File: module/mailnews/mailfile.c [hunk 1]
```diff
@@ -310,8 +310,7 @@ Bool MailParseMessage(int msgnum, MailInfo *info)
    char filename[MAX_PATH + FILENAME_MAX];
    char line[MAX_LINE];
    int num_fields = 5;  // Don't increase this without changing szLoadString (only 5 at a time)
```
Comment (akirmse) [inline module/mailnews/mailfile.c:312]: This looks very suspicious, as there are no longer 5 fields.

File: module/mailnews/mailfile.c [hunk 2]
```diff
@@ -405,8 +405,7 @@ Bool MailParseMessageHeader(int msgnum, char *filename, MailHeader *header)
 {
    FILE *infile;
    int num_fields = 5;  // Don't increase this without changing szLoadString (only 5 at a time)
```
Comment (akirmse) [inline module/mailnews/mailfile.c:407]: Here too.

**Review by akirmse — COMMENTED** [review]
Please fix the PR title.

File: module/mailnews/mailfile.c [hunk 3]
```diff
@@ -161,9 +161,13 @@ void MailNewMessage(int server_index, char *sender, int num_recipients,
...
-         subject_found = False;
-         subject = "";
+      if (strncmp(message, subject_str, strlen(subject_str))) 
+      {
+         subject_str = GetString(hInst, IDS_SUBJECT_PORTUGUESE);
+         if (strncmp(message, subject_str, strlen(subject_str))) {
+            subject_found = False;
+            subject = "";
+         }
       }
```
Comment (akirmse) [inline module/mailnews/mailfile.c:171]: Now that there are 3 of these, let's make this into a loop over the ids instead.  Let me know if you want help with that.

**Review by akirmse — COMMENTED** [review]
Please change the title of this PR to something meaningful.

**Comment by davidklaw** [comment]
In addition to making changes on feedback, I also discovered I hadn't review and addressed issues with the chess and character creation dialogs. Those have been now been reviewed and updated.

File: module/mailnews/mailfile.c [hunk 4]
```diff
@@ -428,12 +439,12 @@ Bool MailParseMessageHeader(int msgnum, char *filename, MailHeader *header)
...
       for (i=0; i < num_fields; i++)
       {
-         if (!strnicmp(line, fields[i], strlen(fields[i])))
+         if (!fields[i].empty() && !strnicmp(line, fields[i].c_str(), strlen(fields[i].c_str())))
          {
             /* Remove newline at end of line */
             line[strlen(line) - 1] = 0;
             
-            ptr = line + strlen(fields[i]);
+            ptr = line + strlen(fields[i].c_str());
```
Comment (akirmse) [inline module/mailnews/mailfile.c:447]: strlen(fields[i].c_str()) -> fields[i].size()

File: module/mailnews/mailfile.c [hunk 5]
```diff
@@ -309,10 +319,10 @@ Bool MailParseMessage(int msgnum, MailInfo *info)
    FILE *infile;
    char filename[MAX_PATH + FILENAME_MAX];
    char line[MAX_LINE];
-   int num_fields = 5;  // Don't increase this without changing szLoadString (only 5 at a time)
-   int field_ids[] = {IDS_SUBJECT_ENGLISH, IDS_SUBJECT_GERMAN,
-                      IDS_FROM, IDS_TO, IDS_DATE};
-   char *fields[5], *ptr;
+   int field_ids[] = {IDS_SUBJECT_ENGLISH, IDS_SUBJECT_GERMAN, IDS_SUBJECT_PORTUGUESE, IDS_FROM, IDS_TO, IDS_DATE};
```
Comment (akirmse) [inline module/mailnews/mailfile.c:322]: field_ids[] should be const

File: module/mailnews/mailfile.c [hunk 6]
```diff
@@ -126,12 +128,15 @@ void MailNewMessage(int server_index, char *sender, int num_recipients,
 {
    int index, num_msgs, msgnum;
    char new_msg[MAXMAIL + 200 + MAX_SUBJECT + MAXUSERNAME * MAX_RECIPIENTS];
-   char *subject, *ptr, *subject_str;
+   int subject_ids[] = {IDS_SUBJECT_ENGLISH, IDS_SUBJECT_GERMAN, IDS_SUBJECT_PORTUGUESE};
```
Comment (akirmse) [inline module/mailnews/mailfile.c:131]: Make const

Comment (davidklaw) [reply module/mailnews/mailfile.c:131]: The likelihood of 3 more language anytime soon seems low, but would you like me to add support for more than 5 strings here?

Comment (akirmse) [reply module/mailnews/mailfile.c:131]: Yeah, we should avoid relying on those terrible static bufferes.

**Comment by davidklaw** [comment]
Pulling it back to draft. Replies are populating the subject field with the date :/

File: module/mailnews/mailfile.c (hunk 6 above)
Comment (davidklaw) [reply module/mailnews/mailfile.c]: I gave it a go.

File: clientd3d/client.rc [hunk 7]
```diff
@@ -18,6 +18,754 @@
...
+            END
+            POPUP "Sele��o de itens"
+            BEGIN
+                MENUITEM "Itens n�o selecionados",       3042
+                MENUITEM "Itens selecionados",     3027
+            END
+            POPUP "Gr�ficos de barra"
+            BEGIN
+                MENUITEM "&Barras normais",              3500
+                MENUITEM "&Barras completas",             3501
```
Comment (akirmse) [inline clientd3d/client.rc:71]: We can't have the same hotkey (B, in this case) for multiple items in the same menu.  I'm afraid this is going to require a thorough audit.

Comment (davidklaw) [reply clientd3d/client.rc]: I've gone through and updated hotkeys by comparing available hotkeys to the English version and ensuring that no duplicate hotkeys exist within the same level of each menu.

**Comment by davidklaw** [comment]
I noticed that Portguguese and German translations more often than not, use the numerical equivalents of constant ids (e.g. `-1` vs. `IDC_STATIC`, `IDCANCEL`, etc) . Should the Portuguese resources use these as well?

Here's what I mean
```
BEGIN
    GROUPBOX        "Confirmação",-1,7,28,228,33
    LTEXT           "Confirmar esta ação irá destruir o seu personagem e criar um novo. É de sua responsabilidade iniciar um novo jogo agora.",-1,7,4,235,18
    LTEXT           "Para confirmação esta ação, insira sua senha:",-1,14,40,106,19
    EDITTEXT        1260,131,42,97,12,ES_PASSWORD | ES_AUTOHSCROLL
    DEFPUSHBUTTON   "Confirmar",1,7,65,107,14
    PUSHBUTTON      "Cancelar",2,120,65,115,14
END
```

```
BEGIN
    GROUPBOX        "Verification",IDC_STATIC,7,38,228,43
    LTEXT           "Performing a reroll will destroy your character, and will create a new character for you to start over.  You are responsible for your own choice to begin the game again.",IDC_STATIC,7,7,228,27
    LTEXT           "To verify that you wish to do this, enter your current account password here.",IDC_STATIC,14,50,114,26
    EDITTEXT        IDC_SUICIDE_PASSWORD,131,52,97,12,ES_PASSWORD | ES_AUTOHSCROLL
    DEFPUSHBUTTON   "Reroll",IDOK,131,74,46,14
    PUSHBUTTON      "Cancel",IDCANCEL,184,74,46,14
END
```

**Comment by Meridian59** [comment]
No, the names are better.  If the #defines for these ever change, all those
translations will break.  I don't know why it was done this way.

**Comment by davidklaw** [comment]
Back to draft as I comb through and updated the use of names.

**Comment by davidklaw** [comment]
This is ready for review. I briefly moved it to draft and back just as a safety measure, but it was unnecessary.

The latest commits address a few misses:
- Fixed the use of constant numerals rather than their names (for Portuguese only)
- Made Portuguese mail and and news lists sortable by removing the nosort flag
- Made small refinements to poor label dimensions and positions in the character creation flow

**Pull request merged**